Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -613,6 +623,35 @@ private extension DiffContainerViewController {
         }
     }
     
+    func fetchWatchAndRollbackStatus() {
+        
+        guard let articleTitle, let wkProject else {
+            return
+        }
+        
+        WKWatchlistService().fetchWatchStatus(title: articleTitle, project: wkProject, needsRollbackRights: true) { result in
+            DispatchQueue.main.async { [weak self] in
+                
+                guard let self else {
+                    return
+                }
+                
+                switch result {
+                case .success(let status):
+                    
+                    let needsWatchButton = !status.watched && FeatureFlags.watchlistEnabled
+                    let needsUnwatchButton = status.watched && FeatureFlags.watchlistEnabled
+                    let needsArticleEditHistoryButton = !(revisionRetrievingDelegate is PageHistoryViewController) && FeatureFlags.watchlistEnabled
```

Comment:
I took liberties in hiding these buttons here if the watchlist feature is disabled. Also, I don't think it makes sense to have an article edit history button if they just came from the article edit history view (they could simply tap Back). My thinking is that this button should only appear if the user came from watchlist. Happy to change it if I'm wrong here though.

Reply:
Update: After looking at https://github.com/wikimedia/wikipedia-ios/pull/4568, I changed this to just show the article edit history button at all times here - my previous logic here caused it to hide when coming from talk > revision history > diff.